packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3923,6 +3944,23 @@ export class ContainerRuntime
 		this.submit({ type, contents }, localOpMetadata);
 	}
 
+	/**
+	 * Api to allow users of runtime to submit ops if required.
+	 * @param type - Type of external op.
+	 * @param contents - contents of the op.
+	 * @param localOpMetadata - localOpMetadata is any.
+	 */
+	public submitExternalOps(
```

Comment:
This mechanism looks suspicious.
You essentially depend now on a chain of these processors for correctness, and there are no protections that all clients will have same code hooks installed to guarantee eventual consistency. For example code does not assert if there is no proper this.externalOpProcessors for an op.
It is also not obvious that resubmit can be simple, i.e. not require rebasing.

I wonder if what you need here is a real DDS and just let DDS (and code associated with it) handle all of that. We are missing ability to install "system" DDSs under root, but we will need to implement such capability in the future (for Tree DDS to rule the world (runtime)).

Basically, we should be very careful to add any kind of custom behavior like that to container runtime - it's already too complicated and has too many custom flows (like ID compressor, that ideally should also be a DDS).